src/internal/connector/onedrive/collection.go

```diff
@@ -139,6 +140,16 @@ func (od *Item) Info() details.ItemInfo {
 //	return od.info.Modified
 //}
 
+// isTimeoutErr is used to determine if the Graph error returned is because of Timeout
```

Comment:
I added the timeout in https://github.com/alcionai/corso/pull/1607 as the MS Graph API was stalling and returning a timeout at times. Not entirely sure what is causing this, but even in the [link mentioned in the pr](https://github.com/alcionai/corso/blob/0bc6b34e044460686eb4db06cd2830ec8af158e3/src/internal/connector/onedrive/collection.go#L211), the conversation ended in adding retries.

Reply:
I see retires there, but not timeout controls.  Are you sure you're not accidentally confusing throttling for timeouts?

Reply:
My bad, what I meant was that I added retries in #1607 for any error returned, but turns out we already retry when throttling happens in the middleware added [here](https://github.com/alcionai/corso/blob/0bc6b34e044460686eb4db06cd2830ec8af158e3/src/internal/connector/graph/service_helper.go#L45), but not for timeout. So, restricting this retry only to timeouts as we don't wanna retry again if we got throttled even after retrying(with backoff) in the middleware.

Reply:
Oh, okay, awesome!  Thanks for the rundown.  I think this can be easily confused in the code.  If you added the same explanation to the `isTimeoutErr` header, it would help clarify that for future devs, too.

Reply:
I had mentioned a bit more where it is being call, but I have reworked the docs a bit to make it more clear.